src/extension/background-script/connectors/eclair.ts

```diff
@@ -61,30 +69,38 @@ class Eclair implements Connector {
     throw new Error("Not yet supported with the currently used account.");
   }
 
-  // not yet implemented
   async getInvoices(): Promise<GetInvoicesResponse> {
-    console.error(
-      `Not yet supported with the currently used account: ${this.constructor.name}`
-    );
-    throw new Error(
-      `${this.constructor.name}: "getInvoices" is not yet supported. Contact us if you need it.`
-    );
+    const response = await this.request("/listinvoices");
+    const invoices: ConnectorInvoice[] = response
+      .map(
+        (invoice: {
+          paymentHash: string;
+          description: string;
+          timestamp: number;
+          amount: number;
+        }) => ({
+          id: invoice.paymentHash,
+          memo: invoice.description,
+          settled: true,
+          settleDate: invoice.timestamp * 1000,
+          totalAmount: invoice.amount / 1000,
+          type: "received",
+        })
+      )
+      .sort((a: ConnectorInvoice, b: ConnectorInvoice) => {
```

Comment:
Should we add some filtering then here (like we do for commando?):

https://github.com/getAlby/lightning-browser-extension/blob/master/src/extension/background-script/connectors/commando.ts#L206

Reply:
I think the API should always return the same kind of data, also across connectors.

Reply:
Eclair doesn't support this kind of filter at the moment.

Reply:
Exactly, the question is whether we should filter it on the client?

Reply:
This call doesn't even tell you if an invoice was settled or not :) 
The alternative is to use the [audit](https://acinq.github.io/eclair/#audit) api endpoint, but that misses the memo field.

Also if we are being realistic, there are very few people that use their Eclair node in this way so I wouldn't bother with too much more effort. In fact, we haven't received  a single complaint in the 1year+ that it was in the extension while it was broken.

Reply:
> This call doesn't even tell you if an invoice was settled or not :)

Oh, okay then. 👍